diff --git a/intern/ghost/intern/GHOST_ISystem.cc b/intern/ghost/intern/GHOST_ISystem.cc
--- a/intern/ghost/intern/GHOST_ISystem.cc
+++ b/intern/ghost/intern/GHOST_ISystem.cc
@@ -14,6 +14,9 @@
 #include "GHOST_ISystem.hh"
 #include "GHOST_SystemHeadless.hh"
 
+#if defined(__ANDROID__)
+#  include "GHOST_SystemAndroid.hh"
+#endif
+
 #if defined(WITH_GHOST_X11) && defined(WITH_GHOST_WAYLAND)
 #  include "GHOST_SystemWayland.hh"
 #  include "GHOST_SystemX11.hh"
@@ -44,7 +47,11 @@ GHOST_TSuccess GHOST_ISystem::createSystem(bool verbose, [[maybe_unused]] bool b
 
   GHOST_TSuccess success;
   if (!system_) {
-#if defined(WITH_HEADLESS)
+#if defined(__ANDROID__)
+    backends_attempted.push_back({"ANDROID"});
+    CLOG_INFO(&LOG, "Create Android system");
+    system_ = new GHOST_SystemAndroid(background);
+#elif defined(WITH_HEADLESS)
     /* Pass. */
 #elif defined(WITH_GHOST_WAYLAND)
 #  if defined(WITH_GHOST_WAYLAND_DYNLOAD)
